rotkehlchen/chain/ethereum/manager.py

```diff
@@ -169,6 +174,31 @@ def _query_web3_get_logs(
     return events
 
 
+def _prepare_arguments(addr: ChecksumEthAddress) -> List[Any]:
+    try:
+        reversed_domain = address_to_reverse_domain(addr)
+    except (TypeError, ValueError) as e:
+        raise InputError(f'Address {addr} has incorrect format or type. {str(e)}') from e
+    normalized_domain_name = normalize_name(reversed_domain)
+    arguments = [normal_name_to_hash(normalized_domain_name)]
+    return arguments
+
+
+def _encode_contract(params: ContractParams) -> str:
+    contract = EthereumContract(address=params.address, abi=params.abi, deployed_block=0)
+    return contract.encode(method_name=params.method_name, arguments=params.arguments)
+
+
+def _decode_contract(params: ContractParams, result_encoded: Any) -> ChecksumEthAddress:
+    contract = EthereumContract(address=params.address, abi=params.abi, deployed_block=0)
+    result = contract.decode(  # pylint: disable=E1136
+        result=result_encoded,
+        method_name=params.method_name,
+        arguments=params.arguments,
+    )[0]
+    return string_to_ethereum_address(result)
```

Comment:
Would be better to use `ens` related names for this functions? Not all calls decode as ethereum address and it might be confusing

Reply:
Yeah, you are right! Adjusted the names, thanks!